### PR TITLE
Dispatch categorical Series ops to Categorical

### PR DESCRIPTION
**Pull request by jbrockmendel**
Moving towards the One Implementation To Rule Them All, this has `Series[timedelta64].__cmp__` ops wrap `TimedeltaIndex.__cmp__` and `Series[category].__cmp__` wrap `Categorical.__cmp__`

Fixes (and tests) incorrectly named results when comparing Series/Index objects.

Ref #19513

### CODE REVIEW

**Comment by jreback** [comment]
I suspect this would close an issue or 2 can you have a look.

File: pandas/tests/series/test_arithmetic.py [hunk 1]
```diff
@@ -43,6 +43,40 @@ def test_ser_flex_cmp_return_dtypes_empty(self, opname):
...
         tm.assert_series_equal(result, Series([1], ['bool']))
 
+    @pytest.mark.parametrize('op', [operator.eq, operator.ne,
+                                    operator.le, operator.lt,
+                                    operator.ge, operator.gt])
+    @pytest.mark.parametrize('names', [(None, None, None),
+                                       ('foo', 'bar', None),
+                                       ('baz', 'baz', 'baz')])
+    def test_ser_cmp_result_names(self, names, op):
+        # datetime64 dtype
```
Comment (jreback) [inline pandas/tests/series/test_arithmetic.py]: did any of these raise / not work before?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_arithmetic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_arithmetic.py]: None of these work at the moment.  They currently retain the series's name.

**Comment by jbrockmendel** [comment]
I didn't see any issues this resolves.  #19513 needs some rulings on whether Categorical behavior should change to match CategoricalIndex or vice-versa.

**Comment by jreback** [comment]
I think #19513 should be fixed first. This will simplify.

**Comment by jbrockmendel** [comment]
> I think #19513 should be fixed first. This will simplify.

OK.  That will affect the edits here a bit but not a whole lot.  Given I'd like to move forward with fixing comp_method_SERIES for the datetime64 case and _dont_ want multiple PRs touching this code, my preference would be to merge this and follow up when #19513 is resolved.

**Comment by jreback** [comment]
> OK. That will affect the edits here a bit but not a whole lot. Given I'd like to move forward with fixing comp_method_SERIES for the datetime64 case and dont want multiple PRs touching this code, my preference would be to merge this and follow up when #19513 is resolved.

no I think this will have to wait. moving around code just to move it around doesn't advance things. I would just rebase this after.

**Comment by jbrockmendel** [comment]
There are non-Categorical-specific parts of this I'd like to push through to avoid re-doing when implementing other dtypes (mainly datetime64).  Plus this fixes+tests some currently-incorrect names.  Would this be easier to finish off if I reverted some/all of the Categorical edits?

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks. pls circle back and see if you can simplify the categorical / CI logic in here (e.g. make consistent / uniform)

**Comment by jbrockmendel** [comment]
> thanks. pls circle back and see if you can simplify the categorical / CI logic in here (e.g. make consistent / uniform)

Yah, this is definitely a goal, is largely dependent on the canonical behavior being clarified in #19513.

**Comment by jreback** [comment]
i would suggest making ci and categorical have a similar behavior is the first step

**Pull request merged**